avm/ptn/lz/sub-vending/tests/e2e/pim-active-role-assignments/pester/main.tests.ps1

```diff
@@ -0,0 +1,63 @@
+param (
+    [Parameter(Mandatory = $false)]
+    [hashtable] $TestInputData = @{}
+)
+
+Describe 'Bicep Landing Zone (Sub) Vending Tests' {
+
+    BeforeAll {
+        $subscriptionId = $TestInputData.DeploymentOutputs.createdSubId.Value
+        $namePrefix = $TestInputData.DeploymentOutputs.namePrefix.Value
+        $serviceShort = $TestInputData.DeploymentOutputs.serviceShort.Value
+        $location = $TestInputData.DeploymentOutputs.resourceLocation.Value
+        Update-AzConfig -DisplayBreakingChangeWarning $false
+        Select-AzSubscription -subscriptionId $subscriptionId
+    }
+
+    Context 'Subscription Tests' {
+        BeforeAll {
+            $sub = Get-AzSubscription -SubscriptionId $subscriptionId -ErrorAction SilentlyContinue
+        }
+
+        It 'Should have a Subscription with the correct name' {
+            $sub.Name | Should -BeLike "dep-sub-blzv-tests-$namePrefix-$serviceShort*"
+        }
+
+        It 'Should have a Subscription that is enabled' {
+            $sub.State | Should -Be 'Enabled'
+        }
+
+        It "Should have a Subscription with a tag key of 'namePrefix' with a value of '$namePrefix'" {
+            $sub.Tags.namePrefix | Should -Be $namePrefix
+        }
+
+        It "Should have a Subscription with a tag key of 'serviceShort' with a value of '$serviceShort'" {
+            $sub.Tags.serviceShort | Should -Be $serviceShort
+        }
+
+        It "Should have a Subscription that is a child of the Management Group with the ID of 'bicep-lz-vending-automation-child'" {
+            $mgAssociation = Get-AzManagementGroupSubscription -SubscriptionId $subscriptionId -GroupId 'bicep-lz-vending-automation-child' -ErrorAction SilentlyContinue
+            $mgAssociation.Id | Should -Be "/providers/Microsoft.Management/managementGroups/bicep-lz-vending-automation-child/subscriptions/$subscriptionId"
+        }
+    }
+
+    Context 'PIM Role-Based Access Control Assignment Tests' {
+        It 'Should Have a PIM Role Assignment for an known AAD Group with an Active Role-Based Access Control Administrator role directly upon the Resource Group' {
+            $iterationCount = 0
+            do {
+                $roleAssignment = Get-AzRoleAssignmentScheduleRequest -Scope "/subscriptions/$subscriptionId/resourceGroups/rsg-$location-net-hs-$namePrefix-$serviceShort" -ErrorAction SilentlyContinue
+                if ($null -eq $roleAssignment) {
+                    Write-Host "Waiting for Resource Group Role Assignments to be eventually consistent... Iteration: $($iterationCount)" -ForegroundColor Yellow
+                    Start-Sleep -Seconds 40
+                    $iterationCount++
+                }
+            } until (
+                $roleAssignment -ne $null -or $iterationCount -ge 10
+            )
+
+            $roleAssignment.PrincipalId | Should -Be '896b1162-be44-4b28-888a-d01acc1b4271'
```

Comment:
Is this always this GUID? Across tenants?

Reply:
No just the AVM tenant as this is the only tenant we can test in

Reply:
Right, but other contributors may not. As the same value is also hardcoded in the `main.test.bicep` file I'd recommend to instead pass the principal ID in via the corresponding CI secret, and here in the `main.test.ps1` script dynamically fetch the principal ID by finding the principal by name, and then taking its Id

Reply:
Resolved